### DISCUSSION TITLE
Nasa Stage 35 simulation

### DISCUSSION THREAD

**Discussion by safdarianebi** [su2code/SU2#2300] [Q&A]
Hi folks,

I hope all is well with you. I am trying to simulate a transonix axial flow compressor, and my simulation will be conducted in specific spans of blade. Therefore, I am trying to conduct a 2-D simulation. 

this is my **config file** information:

SOLVER= RANS
KIND_TURB_MODEL= SST
MATH_PROBLEM= DIRECT
RESTART_SOL= NO
SYSTEM_MEASUREMENTS= SI
MACH_NUMBER= 1.301
AOA= 0.0
SIDESLIP_ANGLE= 0.0
FREESTREAM_PRESSURE= 101325.0
FREESTREAM_TEMPERATURE= 288.15
FREESTREAM_DENSITY= 1.2252487723083554
INIT_OPTION= TD_CONDITIONS
FREESTREAM_OPTION= TEMPERATURE_FS
FREESTREAM_TURBULENCEINTENSITY= 0.05
FREESTREAM_TURB2LAMVISCRATIO= 100.0
REF_ORIGIN_MOMENT_X= 0.0
REF_ORIGIN_MOMENT_Y= 0.0
REF_ORIGIN_MOMENT_Z= 0.0
REF_LENGTH= 1.0
REF_AREA= 1.0
REF_DIMENSIONALIZATION= DIMENSIONAL
FLUID_MODEL= IDEAL_GAS
GAMMA_VALUE= 1.4021650832288652
GAS_CONSTANT= 287.11548963531527
VISCOSITY_MODEL= SUTHERLAND
MU_REF= 1.716e-05
MU_T_REF= 273.15
SUTHERLAND_CONSTANT= 110.4
CONDUCTIVITY_MODEL= CONSTANT_PRANDTL
AVERAGE_PROCESS_KIND= MIXEDOUT
PERFORMANCE_AVERAGE_PROCESS_KIND= MIXEDOUT
MIXEDOUT_COEFF= (1.0, 1.0E-05, 15)
AVERAGE_MACH_LIMIT= 0.05
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= LU_SGS
LINEAR_SOLVER_ERROR= 1e-06
LINEAR_SOLVER_ITER= 10
VENKAT_LIMITER_COEFF= 0.05
LIMITER_ITER= 999999
CONV_NUM_METHOD_FLOW= ROE
MUSCL_FLOW= NO
SLOPE_LIMITER_FLOW= VAN_ALBADA_EDGE
ENTROPY_FIX_COEFF= 0.05
JST_SENSOR_COEFF= (0.5, 0.02)
TIME_DISCRE_FLOW= EULER_IMPLICIT
CONV_NUM_METHOD_TURB= SCALAR_UPWIND
TIME_DISCRE_TURB= EULER_IMPLICIT
CFL_REDUCTION_TURB= 1.0
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES
CFL_NUMBER= 10.0
CFL_ADAPT= YES
CFL_ADAPT_PARAM= (0.1, 2.0, 1, 10)
OUTER_ITER= 15000
CONV_RESIDUAL_MINVAL= -6
CONV_STARTITER= 10
CONV_CAUCHY_ELEMS= 100
CONV_CAUCHY_EPS= 1e-06
MULTIZONE= YES
CONFIG_LIST= (/home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/zone_1_1.cfg,/home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/zone_1_2.cfg)
MARKER_HEATFLUX= ( stage1_wall1, 0.0, stage1_wall2, 0.0 )
MARKER_PERIODIC= ( stage1_periodic1, stage1_periodic2, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.027777777777777776, 0.0, stage1_periodic3, stage1_periodic4, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.021739130434782608, 0.0 )
MARKER_TURBOMACHINERY= ( stage1_inflow, stage1_outmix1, stage1_inmix1, stage1_outflow )
MARKER_MIXINGPLANE_INTERFACE= ( stage1_outmix1, stage1_inmix1 )
MARKER_ZONE_INTERFACE= ( stage1_outmix1, stage1_inmix1 )
MARKER_GILES= ( stage1_inflow, TOTAL_CONDITIONS_PT, 101325.0, 288.15, 1.0, 0.0, 0.0,1.0,1.0, stage1_outmix1, MIXING_OUT, 0.0, 0.0, 0.0, 0.0, 0.0,1.0,1.0, stage1_inmix1, MIXING_IN, 0.0, 0.0, 0.0, 0.0, 0.0,1.0, 1.0, stage1_outflow, STATIC_PRESSURE, 135000.0, 0.0, 0.0, 0.0, 0.0,1.0,1.0 )
TURBOMACHINERY_KIND= AXIAL AXIAL
TURBO_PERF_KIND= COMPRESSOR COMPRESSOR
MARKER_PLOTTING= ( stage1_wall1, stage1_wall2 )
MARKER_ANALYZE= ( stage1_outflow )
TURBULENT_MIXINGPLANE= YES
RAMP_OUTLET_PRESSURE= NO
RAMP_OUTLET_PRESSURE_COEFF= (140000.0, 10.0, 500)
SPATIAL_FOURIER= NO
MESH_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/passage1.su2
MESH_FORMAT= SU2
TABULAR_FORMAT= CSV
SOLUTION_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/restart_flow0.dat
VOLUME_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/flow1.vtu
RESTART_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/restart_flow1.dat
SURFACE_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/surface_flow1.vtu
CONV_FILENAME= /home/ebi/Master-Thesis/Second_Phase/validation/simulation_out/config1.csv
HISTORY_OUTPUT= TURBO_PERF
VOLUME_OUTPUT= (SOLUTION, RESIDUAL, PRIMITIVE)

and simulation condition is as below:

```
flasher = get_flasher("air", "gas")

params=SimulationParams(
    inlet_total_state=flasher.flash(P=101325, T=288.15, mach_number=1.301), 
    target_outlet_static_state=flasher.flash(P=135000, T=323.3),
    translation=[
        np.array([0.0, 286.47, 0.0]),
        None
    ]

```
the biggest problem that I am facing is **divergence of solution for cases with high back pressure.**
Your help and guidance is highly appreciated.

Thanks.

**Comment by vdweide** [comment]
Do you start from scratch in the case of a high back pressure? If yes, that is a well-known issue for computing compressor flows. What you can try is to ramp up the back pressure, possibly via restarts. I.e. compute a solution with e.g. 70% of the back pressure, perform a restart with 80%, etc. In this way you allow the flow to establish itself a bit better.

**Reply by safdarianebi** [reply]
Thank you so much for your reply. That's a good point. I have not tested this yet, but I also tried adjusting config file in this way, which didn't work:

RAMP_OUTLET_PRESSURE= Yes
RAMP_OUTLET_PRESSURE_COEFF= (140000.0, 10.0, 500)

I have doubt about turbulence and viscosity model or cfl number too. Also, the solution converged perfectly with this set up in the case of 125000 Pa back pressure.

**Reply by vdweide** [reply]
Can you restart from the solution with 125000 Pa back pressure and set the back pressure first to 130000 Pa? It may also be that your grid is too coarse to obtain the 140000 Pa.

**Reply by safdarianebi** [reply]
I am trying to validate results for 125000 Pa  back pressure case, and I recognized that the results differ from those obtained by CFX. Therefore, I think I need to adjust my config file. Unfortunately, there is no testcase for transonic compressors and I have doubt about capabilities of su2 in simulating these kinds of cases. However, I hope it can manage it!

Also, you can see the specifications of mesh below:

```
+--------------------------------------------------------------+
|           Mesh Quality Metric|        Minimum|        Maximum|
+--------------------------------------------------------------+
|    Orthogonality Angle (deg.)|        76.9606|             90|
|     CV Face Area Aspect Ratio|         1.0001|        7.56804|
|           CV Sub-Volume Ratio|              1|        6.77479|
+--------------------------------------------------------------+
```
![Screenshot (1428)](https://github.com/su2code/SU2/assets/125588074/386a645d-66d4-4d25-a585-5211e78bb1ea)
